Add blog post about Sphinx ecosystem

astrojuanlu opened the pull request:
This post is more technical than we are used to, but I think it is very useful for our readers and users.

astrojuanlu commented:
🤖 Rendered result: https://readthedocs-blog--150.org.readthedocs.build/sphinx-ecosystem-news-january-2022/

astrojuanlu commented:
Since I quoted @choldgraf twice, pinging him in case he wants to have a look :)

ericholscher commented on sphinx-ecosystem-news-january-2022.rst:
```diff
@@ -0,0 +1,115 @@
+and `custom SVG
+icons <https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#local-image-icons>`_.
+
+Extensions
```
These feel like good tweet content, and we reframe the blog post as annotated sphinx release notes?

astrojuanlu replied:
Not sure if you're advocating for removing this from the blog post and putting it in a Twitter thread, or just restructuring it a bit?

ericholscher replied:
Restructuring the post, but also using this content as a tweet. If we get better at tracking extensions, I think it's great twitter content, especially with the GIF's.

ericholscher reviewed (commented):
This is a great concept, unfortunately we don't have a plan to continue publishing something like this, so we need to reframe it as something that isn't the start of a series. I'd advocate for changing this to annotated Sphinx release notes, with a small section highlighting extensions or something. I feel like all of this would be good tweet content, especially the gif's, but I think we should also be focused on RTD some (do we support 4.4? How does a user enable it? Any caveats?)

I love the idea of being advocates for the Sphinx ecosystem, but we need a process in place before we start tracking all extensions and committing to writing ongoing content. I think Sphinx release notes feels like the right amount we can commit to, and something that is core to our users.

ericholscher commented on sphinx-ecosystem-news-january-2022.rst:
```diff
@@ -0,0 +1,115 @@
+`sphinx-codeautolink 0.9 <https://sphinx-codeautolink.readthedocs.io/en/stable/release_notes.html#id2>`_
+   sphinx-codeautolink is a Sphinx extension to automatically include links inside code blocks.
+   The 0.9 version now links Python builtin objects if intersphinx is properly configured
+   and has several logging improvements.
```
We should be promoting this extension more! It seems super useful.

astrojuanlu replied:
Yep: https://github.com/humitos/sphinx-extensions/issues/18

choldgraf replied:
cool - I hadn't heard about this before either!

ericholscher commented on sphinx-ecosystem-news-january-2022.rst:
```diff
@@ -0,0 +1,115 @@
+ 
+   .. code-block:: rst
+ 
+      Choose menu item :constdata:label:`menu.yaml?FileSaveAs`.
```
Huh, interesting UX.

ericholscher commented on sphinx-ecosystem-news-january-2022.rst:
```diff
@@ -0,0 +1,115 @@
+   :align: center
+   :alt: Preview of MyST integrated in Jupyter notebooks.
+
+   Preview of MyST integrated in Jupyter notebooks.
```
This is an abrupt ending... We should close this out with something.

astrojuanlu replied:
Fair enough 😅

ericholscher reviewed (commented):
A couple more content notes:

astrojuanlu commented:
> I'd advocate for changing this to annotated Sphinx release notes, with a small section highlighting extensions or something.
> ...I think Sphinx release notes feels like the right amount we can commit to, and something that is core to our users.

Sounds good, I'll reframe the content and title a bit. Left a comment in the relevant thread.

> I think we should also be focused on RTD some (do we support 4.4? How does a user enable it? Any caveats?)

Good point, will add part of that too as well.

choldgraf reviewed (commented):
looks good to me in general - thanks for the shout-out!

astrojuanlu commented:
Restructured the post a bit: changed the title to make it more focused on Sphinx 4.4, grouped all extensions and themes inside the same section, and added a paragraph on how to use this on Read the Docs.

ericholscher reviewed (approved):
Changes look good 👍

humitos commented on sphinx-4-4-release-other-ecosystem-news.rst:
```diff
@@ -0,0 +1,133 @@
+Using Sphinx 4.4 on Read the Docs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sphinx 4.4 is supported on Read the Docs. By default, RTD will install the latest version for you,
+as described in :ref:`our documentation <readthedocs:builds:external dependencies>`.
```
I'm happy this is very clear now in our documentation, it's easy to explain to user ("we install the latest version available"), and there are no surprises :100:. Way better than pinning to an old version and having to explain the "why" to users.

humitos reviewed (approved):
Very good blogpost! It would be awesome if we can sustain this over time!

astrojuanlu commented:
Thanks both for the review! Merging and tweeting 💪🏽
